api/bag_vrrefinements.cpp

```diff
@@ -151,6 +152,12 @@ std::unique_ptr<VRRefinements> VRRefinements::open(
         new ::H5::DataSet{h5file.openDataSet(VR_REFINEMENT_PATH)},
             DeleteH5dataSet{});
 
+    hsize_t dims[2];
+    int ndims = h5dataSet->getSpace().getSimpleExtentDims(dims, nullptr);
+    if (ndims != 2) {
+        throw InvalidVRRefinementDimensions{};
```

Comment:
@brian-r-calder Here we are checking for there to be two dimensions, if there aren't, an `InvalidVRRefinementDimensions` exception is thrown. However, the message from `InvalidVRRefinementDimensions` currently reads: "The variable resolution refinement layer is not 1 dimensional.", which is confusing. Can you clarify?

Reply:
This is a case of adapting to the reality of BAG files in the field.  Although the specification says that VR refinements is a one-dimensional array, and should be written into the HDF5 file as a single dimensional dataset, BAG files retrieved from NCEI (from NOAA surveys) do not have this form.  Although not entirely proven, I suspect that this is because the GDAL drivers are not following the specification as written.

We have a couple of options here: we can attempt to get the GDAL drivers re-written and then re-write ever BAG file in the US national archive (and presumably everywhere else that BAG files are created using GDAL -- meaning everything that uses CARIS software, most likely); or, we can adapt the reference library to match the field experience.  Given the level of complexity in the former, I've chosen to pursue the latter.

To fix this, I've changed the exception to say that the dimensions are "inconsistent with specification", but this also means that we'll have to change the test to reflect this.